File: 1599/solution.go
package _1599

func minOperationsMaxProfit(customers []int, boardingCost int, runningCost int) int {
	if 4*boardingCost-runningCost <= 0 {
		return -1
	}
	var minRound, curRound, maxProfits, curProfits, restNum int
	i, n := 0, len(customers)
	for i < n || restNum != 0 {
		if i < n {
			restNum += customers[i]
			i++
		}
		curProfits += min(restNum, 4)*boardingCost - runningCost
		curRound++
		if curProfits > maxProfits {
			maxProfits = curProfits
			minRound = curRound
		}
		restNum -= min(restNum, 4)
	}
	if minRound > 0 {
		return minRound
	}
	return -1
}

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}

File: 1599/Solution.java
class Solution {
    public int minOperationsMaxProfit(int[] customers, int boardingCost, int runningCost) {
        if (4 * boardingCost - runningCost <= 0) {
            return -1;
        }
        int minRound = 0, curRound = 0, maxProfits = 0, curProfits = 0, restNum = 0;
        int i = 0, n = customers.length;
        while (i < n || restNum > 0) {
            if (i < n) {
                restNum += customers[i];
                i++;
            }
            curProfits += Math.min(restNum, 4) * boardingCost - runningCost;
            curRound++;
            if (curProfits > maxProfits) {
                minRound = curRound;
                maxProfits = curProfits;
            }
            restNum -= Math.min(restNum, 4);
        }
        return minRound > 0 ? minRound : -1;
    }
}